### ISSUE TITLE
How to set text alignment within plot title

### ISSUE THREAD

**Issue by francisbarton** [mtennekes/tmap#539]
Is there an option for this? I haven't found one.

I have `title.position = c("RIGHT", "TOP")` but the title text itself is left-aligned within the box, and is near the centre of the plot rather than to the right hand side.
A layout option to set title text alignment would be great. Or maybe there is already a away to do this.

**Comment by Rapsodia86** [comment]
Fine tunning helped:
`tm_shape(nz) +
  tm_polygons() +
  tm_title("My title", position = tm_pos_out("center", "top", pos.h = 0.71, pos.v = 1))`

![Image](https://github.com/user-attachments/assets/c3f2cced-f7ff-42c0-881d-6a04ae3dbc1f)

And here is a question:
I wanted to set pos.h as numeric (for fine-tuning) while pos.v leave as "right" and that gave me an error:
Error: position argument: for pos.h and .v should both be numeric or character values.
Would it be possible to mix numeric and character values?

Thanks!

**Comment by mtennekes** [comment]
> [@mtennekes](https://github.com/mtennekes) is the following code the best approach to move the title to top right?
> 
> tm_shape(nz) +
>   tm_polygons() +
>   tm_title("My title", position = tm_pos_out("center", "top", pos.h = "right"))

Yes, that is indeed the correct (i.e. intended) approach. It doesn't mean that it's as-user-friendly/intuitive as possible, but at least it works consistently (I hope) and is flexible. 

Of course, we could think about improving the usability, for instance adding more shortcut-functions:`tm_pos_above`, `tm_pos_left`, `tm_pos_right`, `tm_pos_bottom`. Then your example would be:

```{r}
 tm_shape(nz) +
   tm_polygons() +
   tm_title("My title", position = tm_pos_top(pos.h = "right"))
```


> Fine tunning helped:
> tm_shape(nz) + tm_polygons() + tm_title("My title", position = tm_pos_out("center", "top", pos.h = 0.71, pos.v = 1))


Also the intended use! Unfortunately it's not possible to mix numbers with characters, because with characters the position of the legends (and other map components) are processed in a smart (and complex) way, taking into account margins, overlap, stacking etc. With numbers these aspected are ignore in favour of freedom.

We could think of a way to mix characters and numbers, e.g. by assigning each character value with a number, say left -> 0.05, right -> 0.95.
